### PR TITLE
add task solution - Layout-html-form

### PR DESCRIPTION
**Pull request by Sadeni-09**
- [DEMO LINK](https://Sadeni-09.github.io/layout_html-form/)
- [TEST REPORT LINK](https://Sadeni-09.github.io/layout_html-form/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,17 +1,152 @@
...
     <link rel="stylesheet" href="./style.css">
   </head>
+
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post"></form>
+    <fieldset>
+      <legend>Information:</legend>
+      <div class="form-field">
```
Comment (anstsot) [inline src/index.html]: add empty row between siblings
```suggestion
      <legend>Information:</legend>
      
      <div class="form-field">
```

File: src/index.html [hunk 2]
```diff
@@ -1,17 +1,152 @@
...
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>HTML Form</title>
     <link rel="stylesheet" href="./style.css">
   </head>
+
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post"></form>
+    <fieldset>
```
Comment (anstsot) [inline src/index.html]: the form should contain all fields
```suggestion
    <form action="http://localhost:8080/api" method="post">
      <fieldset>
```

File: src/index.html [hunk 3]
```diff
@@ -1,17 +1,152 @@
...
+      <legend>Information:</legend>
+      <div class="form-field">
+        <label for="name">Name:
+          <input type="text" 
+        name="name" 
+        required 
+        autocomplete="off"
+          >
```
Comment (anstsot) [inline src/index.html]: check the spacings
```suggestion
          <input
            type="text" 
            name="name" 
            required 
            autocomplete="off"
          >
```

File: src/index.html [hunk 4]
```diff
@@ -1,17 +1,152 @@
...
   </head>
+
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post"></form>
+    <fieldset>
+      <legend>Information:</legend>
+      <div class="form-field">
+        <label for="name">Name:
```
Comment (anstsot) [inline src/index.html]: the child should start from new row
```suggestion
        <label for="name">
          Name:
```

File: src/style.css [hunk 5]
```diff
@@ -1 +1,21 @@
 /* styles go here */
+fieldset {
+  display: block;
+  border-style: solid 5px black;
+  margin: 15px 0;
+}
+
+input {
+  display: inline;
+  margin: 10px 0;
```
Comment (anstsot) [inline src/style.css]: check the margins. you only need `margin-bottom`

**Review by anstsot — CHANGES_REQUESTED** [review]
Good work. But please fix the issues according to the [checklist](https://github.com/mate-academy/layout_html-form/blob/master/checklist.md) and try to pass all tests.

File: src/index.html [hunk 6]
```diff
@@ -1,17 +1,166 @@
...
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post">
+      <fieldset>
+        <legend>Personal information:</legend>
+        <div class="form-field">
+          <label for="surname">Surname:
+            <input type="text" 
+                name="surname" 
+                autocomplete="off"
+            >
```
Comment (witflash) [inline src/index.html]:
```suggestion
            <input 
              type="text" 
              name="surname" 
              autocomplete="off"
            >
```

File: src/index.html [hunk 7]
```diff
@@ -1,17 +1,166 @@
...
+          </label>
+        </div>
+
+        <div class="form-field">
+          <label for="name">Name:
+            <input type="text" 
+            name="name" 
+            required 
+            autocomplete="off"
+            >
```
Comment (witflash) [inline src/index.html]:
```suggestion
            <input 
              type="text" 
              name="name" 
              required 
              autocomplete="off"
            >
```

File: src/index.html [hunk 8]
```diff
@@ -1,17 +1,166 @@
...
+        </div>
+      
+        <div class="form-field">
+          <label for="age">How old are You?
+            <input type="number" 
+            name="age" 
+            value="12" 
+            min="1" 
+            max="100"
+            >
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            How old are You?
            <input 
              type="number" 
              name="age" 
              value="12" 
              min="1" 
              max="100"
            >
```

File: src/index.html [hunk 9]
```diff
@@ -1,17 +1,166 @@
...
+          </label>
+        </div>
+      
+        <div class="form-field">
+          <label for="dateOfBirth">Full date of birth:
+            <input type="date" 
+            name="dateOfBirth" 
+            value="mm/dd/yyyy"
+            >
+          </label>
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            Full date of birth:
            <input 
              type="date" 
              name="dateOfBirth" 
              value="mm/dd/yyyy"
            >
          </label>
```

File: src/index.html [hunk 10]
```diff
@@ -1,17 +1,166 @@
...
+        </div>
+      
+        <div class="terms">
+          <label for="terms">I accept the term of the agreement
+            <input type="checkbox" name="terms">
+          </label>
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            I accept the term of the agreement
            <input type="checkbox" name="terms">
          </label>
```

File: src/index.html [hunk 11]
```diff
@@ -1,17 +1,166 @@
...
+
+      <fieldset>
+        <legend>Registration:</legend>
+        <div class="form-field">
+          <label for="email">E-mail:
+            <input type="email" 
+            name="email" 
+            placeholder="email@example.com" 
+            required
+            >
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            E-mail:
            <input 
              type="email" 
              name="email" 
              placeholder="email@example.com" 
              required
            >
```

File: src/index.html [hunk 12]
```diff
@@ -1,17 +1,166 @@
...
+          </label>
+        </div>
+      
+        <div class="password">
+          <label for="password">Password:
+            <input type="password" 
+            name="password" 
+            required
+            >
+          </label>
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            Password:
            <input 
              type="password" 
              name="password" 
              required
            >
          </label>
```

File: src/index.html [hunk 13]
```diff
@@ -1,17 +1,166 @@
...
+        <legend>An interesting fact about you!</legend>
+        <div class="form-field">
+      Do you love cats?
+          <label>
+            <input type="radio" name="cat">Yes
+          </label>
+          <label>
+            <input type="radio" name="cat">No
+          </label>
+        </div>
```
Comment (witflash) [inline src/index.html]:
```suggestion
        <div class="form-field">
          Do you love cats?
          <label>
            <input type="radio" name="cat">
            Yes
          </label>
          <label>
            <input type="radio" name="cat">
            No
          </label>
        </div>
```

File: src/index.html [hunk 14]
```diff
@@ -1,17 +1,166 @@
...
+      
+        <div class="form-field">
+          <label for="color">What is your favorite color?
+            <input type="color" 
+            name="color" 
+            value="#000000"
+            >
```
Comment (witflash) [inline src/index.html]:
```suggestion
            <input 
              type="color" 
              name="color" 
              value="#000000"
            >
```

File: src/index.html [hunk 15]
```diff
@@ -1,17 +1,166 @@
...
+          </label>
+        </div>
+
+        <div class="form-field">
+          <label for="timeToBed">What time do you go to bed?
+            <input type="time" 
+            name="timeToBed" 
+            step="2"
+            >
+          </label>
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            What time do you go to bed?
            <input 
              type="time" 
              name="timeToBed" 
              step="2"
            >
          </label>
```

File: src/index.html [hunk 16]
```diff
@@ -1,17 +1,166 @@
...
+      
+        <div class="work">
+          <label for="rate">How do you rate our work?
+            <input type="range" 
+            name="rate" 
+            min="0" 
+            max="100"
+            value="0"
+            >
+          </label>
```
Comment (witflash) [inline src/index.html]:
```suggestion
          <label>
            How do you rate our work?
            <input 
              type="range" 
              name="rate" 
              min="0" 
              max="100"
              value="0"
            >
          </label>
```

File: src/index.html [hunk 17]
```diff
@@ -1,17 +1,166 @@
...
+        <legend>Additional info:</legend>
+        <div class="form-field">
+          <label for="comments">Comments:
+            <textarea class="comment" 
+          name="comments" 
+          minlength="50" 
+          maxlength="300"
+          >
+            </textarea>
+          </label>
```
Comment (witflash) [inline src/index.html:166]:
```suggestion
          <label>
            Comments:
            <textarea 
              class="comment" 
              name="comments" 
              minlength="50" 
              maxlength="300"
            >
            </textarea>
          </label>
```

File: src/index.html (hunk 9 above)
Comment (witflash) [reply src/index.html]: You may avoid using `for` attribule if your `label` is a wrapper for `input`.

**Review by witflash — CHANGES_REQUESTED** [review]
Well done! 👍 
Let's make the last syntax changes.

File: src/index.html (hunk 9 above)
Comment (Sadeni-09) [reply src/index.html]: I didn't understand this, sorry. Where is "for"

File: src/index.html [hunk 18]
```diff
@@ -1,17 +1,184 @@
...
     >
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>HTML Form</title>
     <link rel="stylesheet" href="./style.css">
   </head>
+
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post">
```
Comment (Anastasiia-Svintsova) [inline src/index.html:16]: Follow the requirements, please. Don't use localhost in action
image

File: src/index.html [hunk 19]
```diff
@@ -1,17 +1,184 @@
...
+      <fieldset>
+        <legend>Personal information:</legend>
+        <div class="form-field">
+          <label for="surname">Surname:
```
Comment (Anastasiia-Svintsova) [inline src/index.html:20]: Check the code style everywhere 
```suggestion
          <label for="surname">
            Surname:
```

File: src/index.html [hunk 20]
```diff
@@ -1,17 +1,184 @@
...
+          <label>
+            E-mail:
+            <input 
+              type="email" 
+              name="email" 
+              placeholder="email@example.com" 
+              required
+            >
+          </label>
+        </div>
```
Comment (Anastasiia-Svintsova) [inline src/index.html:84]: Why do you need additional div for each input? Remove them
```suggestion
          <label class="form-field">
            E-mail:
            <input 
              type="email" 
              name="email" 
              placeholder="email@example.com" 
              required
            >
          </label>
```

File: src/index.html [hunk 21]
```diff
@@ -1,17 +1,184 @@
...
+              type="range" 
+              name="rate" 
+              min="0" 
+              max="100"
+              value="0"
+            >
+          </label>
+        </div>
+      </fieldset>
+      <fieldset>
```
Comment (Anastasiia-Svintsova) [inline src/index.html:156]: Don't forget to add empty lines between multiline sibling blocks of HTML
```suggestion
      </fieldset>
      
      <fieldset>
```

File: src/index.html [hunk 22]
```diff
@@ -1,17 +1,184 @@
...
+        <legend>Additional info:</legend>
+        <div class="form-field">
+          <label for="comments">Comments:
+            <textarea class="comment" 
+          name="comments" 
+          minlength="50" 
+          maxlength="300"
+          >
+            </textarea>
```
Comment (Anastasiia-Svintsova) [inline src/index.html:165]: Follow the code style from the checklist, please 
image

File: src/index.html [hunk 23]
```diff
@@ -1,17 +1,184 @@
...
+            <select name="reccomend">
+              <option value="yes">yes</option>
+              <option value="no">no</option>
+            </select>
+          </label>
+        </div>
+      </fieldset>
+      <div>
+        <button type="submit">Submit</button>
+      </div>
```
Comment (Anastasiia-Svintsova) [inline src/index.html:180]: Same. Don't add additional wrappers everywhere 
```suggestion
        <button type="submit">Submit</button>
```

File: src/style.css [hunk 24]
```diff
@@ -1 +1,8 @@
 /* styles go here */
+fieldset {
```
Comment (Anastasiia-Svintsova) [inline src/style.css:2]: Don't style tags, use class selector instead

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]

**Pull request closed without merging**